### PR TITLE
chore(deps): update to rc.1

### PR DESCRIPTION
**Pull request by mgechev**

### CODE REVIEW

**Comment by NathanWalker** [comment]
I think `test-main.js` may need some mapping/aliasing... something like (disregard helpers part of course)?:
screen shot 2016-05-03 at 9 40 36 am

**Comment by JohnCashmore** [comment]
@mgechev you want any help with this? or is it nearly finished?

**Comment by mgechev** [comment]
I'm quite busy around ng-conf this week so help will be greatly appreciated :-).

**Comment by mgechev** [comment]
@JohnCashmore the migration should be completed. Would you review the changes?

**Comment by JohnCashmore** [comment]
I'll pull it down and have a look through

**Comment by JohnCashmore** [comment]
There is an issue with the service worker dependency, which means the check versions fails on a clean `npm install`

```
npm ERR! peerinvalid The package es6-shim@0.33.13 does not satisfy its siblings' peerDependencies requirements!
npm ERR! peerinvalid Peer angular2@2.0.0-beta.17 wants es6-shim@^0.35.0
```

And there is an issue with the router.  on one of the specs, i'll try and have a look this evening or in the morning.

**Comment by mgechev** [comment]
I didn't get any peerDependency errors. Did you run your test over the rc0 branch?

**Comment by mgechev** [comment]
@JohnCashmore had issues with the production build, but everything should be fine now.

**Comment by mgechev** [comment]
Btw, RootRoute is not exported so the typescript compiler may complain. Here's a fix https://github.com/angular/angular/pull/8450

**Comment by JohnCashmore** [comment]
I did do it I this branch, perhaps it's a issue local to me, l'll try again shortly

**Comment by JohnCashmore** [comment]
This package https://www.npmjs.com/package/angular2-service-worker is causing me the problems in the package.json it references a beta version of angular and i think it needs to be either updated or replaced, you'll know more then me. I'd suggest the service worker is removed from the seed for now until theres a solid solution.

**Comment by mgechev** [comment]
Thanks for the update! Lets ping @jeffbcross.

File: tools/config/seed.config.ts [hunk 1]
```diff
@@ -94,7 +92,10 @@ export class SeedConfig {
 
   SYSTEM_BUILDER_CONFIG = {
     defaultJSExtensions: true,
-    packageConfigPaths: [join(this.PROJECT_ROOT, 'node_modules', '*', 'package.json')],
+    packageConfigPaths: [
+      join(this.PROJECT_ROOT, 'node_modules', '*', 'package.json'),
+      join(this.PROJECT_ROOT, 'node_modules', '@angular', '*', 'package.json')
```
Comment (ludohenin) [inline tools/config/seed.config.ts]: Shouldn't this be the same as above (L77-78) ?

**Comment by JohnCashmore** [comment]
The site also doesn't work for me i am getting the error below in the console, which is weird 

```
localhost/:86 Error: Error: XHR error (404 Not Found) loading http://localhost:5555/node_modules/@angular/package.json(…) "Report this error at https://github.com/mgechev/angular2-seed/issues"
```

**Comment by daniru** [comment]
I made it run changing updating seed.config.ts

 packageConfigPaths: [
      `${this.APP_BASE}node_modules/@angular/*/package.json`,
      `${this.APP_BASE}node_modules/*/package.json`,
      `${this.APP_BASE}node_modules/**/package.json`
    ],

**Comment by Benjman** [comment]
Getting this error trying to do a completely fresh clone, install and run of this branch:

`Error: patchProperty/desc.set/wrapFn@http://localhost:5555/node_modules/zone.js/dist/zone.js?1462479592560:769:27 Zone</ZoneDelegate</ZoneDelegate.prototype.invokeTask@http://localhost:5555/node_modules/zone.js/dist/zone.js?1462479592560:356:24 Zone</Zone</Zone.prototype.runTask@http://localhost:5555/node_modules/zone.js/dist/zone.js?1462479592560:256:29 ZoneTask/this.invoke@http://localhost:5555/node_modules/zone.js/dist/zone.js?1462479592560:423:29 Error loading http://localhost:5555/node_modules/@angular/package.json Error loading http://localhost:5555/main.js
Report this error at https://github.com/mgechev/angular2-seed/issues`

NPM version 3.8.6
Node version v5.11.0

Attached is the output after running
`git clone -b rc0 https://github.com/mgechev/angular2-seed.git && cd angular2-seed/ && npm i && npm start`
[angular2-seed_clone_output.txt](https://github.com/mgechev/angular2-seed/files/251179/angular2-seed_clone_output.txt)

**Comment by bonejon** [comment]
I'm getting the following error when I run `npm run build.prod`

`[10:06:39] Starting 'build.bundles.app'...
Unhandled rejection Error on fetch for @angular/core.js at file:///C:/Users/jon.
bone/Documents/angular2-seed/node_modules/@angular/core.js
        Loading dist/tmp/main.js
        Error: ENOENT: no such file or directory, open 'C:\Users\jon.bone\Docume
nts\angular2-seed\node_modules\@angular\core.js'
    at Error (native)`

I'll post up if I work out a solution.

Thanks, Jon.

**Comment by ZuSe** [comment]
@daniru:
When I change it as you suggested i get:

> es6-shim.js:1515:11
> Error: BaseException@http://localhost:5555/node_modules/@angular/core/src/facade/exceptions.js:17:23
> 	InvalidProviderError@http://localhost:5555/node_modules/@angular/core/src/di/reflective_exceptions.js:180:9
> 	_normalizeProviders/<@http://localhost:5555/node_modules/@angular/core/src/di/reflective_provider.js:162:19
> 	Call@http://localhost:5555/node_modules/es6-shim/es6-shim.js?1462547326130:279:14
> 	forEach@http://localhost:5555/node_modules/es6-shim/es6-shim.js?1462547326130:1269:14
> 	_normalizeProviders@http://localhost:5555/node_modules/@angular/core/src/di/reflective_provider.js:145:5
> 	_normalizeProviders/<@http://localhost:5555/node_modules/@angular/core/src/di/reflective_provider.js:156:13
> 	Call@http://localhost:5555/node_modules/es6-shim/es6-shim.js?1462547326130:279:14
> 	forEach@http://localhost:5555/node_modules/es6-shim/es6-shim.js?1462547326130:1269:14
> 	_normalizeProviders@http://localhost:5555/node_modules/@angular/core/src/di/reflective_provider.js:145:5
> 	_normalizeProviders/<@http://localhost:5555/node_modules/@angular/core/src/di/reflective_provider.js:156:13
> 	Call@http://localhost:5555/node_modules/es6-shim/es6-shim.js?1462547326130:279:14
> 	forEach@http://localhost:5555/node_modules/es6-shim/es6-shim.js?1462547326130:1269:14
> 	_normalizeProviders@http://localhost:5555/node_modules/@angular/core/src/di/reflective_provider.js:145:5
> 	resolveReflectiveProviders@http://localhost:5555/node_modules/@angular/core/src/di/reflective_provider.js:103:22
> 	ReflectiveInjector.resolve@http://localhost:5555/node_modules/@angular/core/src/di/reflective_injector.js:357:16
> 	ReflectiveInjector.resolveAndCreate@http://localhost:5555/node_modules/@angular/core/src/di/reflective_injector.js:387:43
> 	bootstrap@http://localhost:5555/node_modules/@angular/platform-browser-dynamic/platform_browser_dynamic.js:90:23
> 	@http://localhost:5555/main.js:11:1
> 	@http://localhost:5555/main.js:1:31
> 	@http://localhost:5555/main.js:1:2
> 	Zone</ZoneDelegate</ZoneDelegate.prototype.invoke@http://localhost:5555/node_modules/zone.js/dist/zone.js?1462547326132:323:20
> 	Zone</Zone</Zone.prototype.run@http://localhost:5555/node_modules/zone.js/dist/zone.js?1462547326132:216:25
> 	scheduleResolveOrReject/<@http://localhost:5555/node_modules/zone.js/dist/zone.js?1462547326132:571:53
> 	Zone</ZoneDelegate</ZoneDelegate.prototype.invokeTask@http://localhost:5555/node_modules/zone.js/dist/zone.js?1462547326132:356:24
> 	Zone</Zone</Zone.prototype.runTask@http://localhost:5555/node_modules/zone.js/dist/zone.js?1462547326132:256:29
> 	drainMicroTaskQueue@http://localhost:5555/node_modules/zone.js/dist/zone.js?1462547326132:474:26
> 	ZoneTask/this.invoke@http://localhost:5555/node_modules/zone.js/dist/zone.js?1462547326132:426:22
> 	
> 	Evaluating http://localhost:5555/main.js
> 	Error loading http://localhost:5555/main.js "Report this error at https://github.com/mgechev/angular2-seed/issues"


Firefox 46

**Comment by mgechev** [comment]
@JohnCashmore your changes are incorporated here. Lets see if the build will pass.

**Comment by NathanWalker** [comment]
Will do, thanks, I'll keep an eye on when the build finishes.

**Comment by mgechev** [comment]
Tests are not failing but there's an exception that needs to be fixed..

**Comment by NathanWalker** [comment]
running through tests now to see the exception.

**Comment by mgechev** [comment]
@NathanWalker should be fine now. Need to go, almost missed my flight.

**Comment by NathanWalker** [comment]
Don't you miss that flight! Ok, I'll look through everything and if all is clear, I'll merge.

**Comment by NathanWalker** [comment]
Squashed and also fixed a small regression with 404 warning from .svg files in unit tests.
The api doc generation is broken but has been for awhile, discussion going on here: https://github.com/NathanWalker/angular2-seed-advanced/issues/53  I'll start on a PR for that tonight.

**Pull request merged**